File: sample/src/main/java/ru/terrakok/cicerone/sample/subnavigation/LocalCiceroneHolder.java
package ru.terrakok.cicerone.sample.subnavigation;

import com.github.terrakok.cicerone.Cicerone;
import com.github.terrakok.cicerone.Router;

import java.util.HashMap;

/**
 * Created by terrakok 27.11.16
 */
public class LocalCiceroneHolder {
    private HashMap<String, Cicerone<Router>> containers;

    public LocalCiceroneHolder() {
        containers = new HashMap<>();
    }

    public Cicerone<Router> getCicerone(String containerTag) {
        if (!containers.containsKey(containerTag)) {
            containers.put(containerTag, Cicerone.Companion.create());
        }
        return containers.get(containerTag);
    }
}

File: sample/src/main/java/ru/terrakok/cicerone/sample/subnavigation/LocalCiceroneHolder.kt
package ru.terrakok.cicerone.sample.subnavigation

import com.github.terrakok.cicerone.Cicerone
import com.github.terrakok.cicerone.Cicerone.Companion.create
import com.github.terrakok.cicerone.Router
import java.util.*

/**
 * Created by terrakok 27.11.16
 */
class LocalCiceroneHolder {
    private val containers = HashMap<String, Cicerone<Router>>()

    fun getCicerone(containerTag: String): Cicerone<Router> =
            containers.getOrPut(containerTag) {
                create()
            }
}